src/libxtp/pmdecomposition.cc

```diff
@@ -0,0 +1,123 @@
+/*
+ *            Copyright 2009-2020 The VOTCA Development Team
+ *                       (http://www.votca.org)
+ *
+ *      Licensed under the Apache License, Version 2.0 (the "License")
+ *
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *              http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include "votca/xtp/pmdecomposition.h"
+#include "votca/xtp/aomatrix.h"
+#include <votca/tools/eigenio_matrixmarket.h>
+
+namespace votca {
+namespace xtp {
+void PMDecomposition::computePMD(Orbitals &orbitals) {
+  Eigen::MatrixXd mo_coeff = orbitals.MOs().eigenvectors().leftCols(
+      orbitals.getNumberOfAlphaElectrons());
+  QMMolecule mol = orbitals.QMAtoms();
+  basis.Load(orbitals.getDFTbasisName());
+  aobasis.Fill(basis, mol);
+  AOOverlap overlap;
+  overlap.Fill(aobasis);
+  Eigen::MatrixXd S = overlap.Matrix();
+  double diff_D = 10000;
+  Index i = 1;
+  while (diff_D > 1e-6) {
+    XTP_LOG(Log::error, log) << "Iteration: " << i << std::flush;
+    Eigen::MatrixXd uppertriangular = orbitalselections(mo_coeff, S);
+    Index maxrow, maxcol;
+    diff_D = uppertriangular.maxCoeff(&maxrow, &maxcol);
+    XTP_LOG(Log::error, log) << maxrow << " " << maxcol << std::flush;
+    XTP_LOG(Log::error, log) << diff_D << std::flush;
+    Eigen::MatrixXd max_orbs(mo_coeff.rows(), 2);
+    max_orbs << mo_coeff.col(maxrow), mo_coeff.col(maxcol);
+    Eigen::MatrixXd new_orbs(mo_coeff.rows(), 2);
```

Comment:
Preferably these should have a datatype Eigen::MatrixX2d

Reply:
Done